packages/data/lib/repository/data_repository.dart

```diff
@@ -1,17 +1,19 @@
+import 'package:dio/dio.dart';
 import 'package:domain/repository/domain_repository.dart';
 
-class PalindromeRepositoryImpl implements PalindromeRepository {
+import '../mapper/mapper.dart';
+
+class PalindromeRepositoryImpl implements PalindromeNetworkRepository {
+  PalindromeRepositoryImpl( this._client);
+
+  final Dio _client;
+
   @override
-  bool checkPalindrome(String text, int first, int last) {
-    if (text.isEmpty | text.contains(RegExp(r'[0-9,^\s*]'))) {
-      return false;
-    } else if (first == last - 1) {
-      return true;
-    } else if (text[first] != text[last - 1]) {
-      return false;
-    } else if (first < last - 1) {
-      return checkPalindrome(text, first + 1, last - 1);
-    }
-    return true;
+  Future<bool> checkPalindrome(String text, int first, int last) async {
+    try {
+      final response = await _client.get(text);
+      return Mapper.call(response.data);
```

Comment:
don't use static, pass instance of mapper in the same way as u pass repository - via di

Reply:
fixed